var/spack/repos/builtin/packages/py-hail/package.py

```diff
@@ -0,0 +1,234 @@
+        # which will not be available in a spack cache. Since we know it from
+        # the package, we can inject it in the failure and move forward.
+        revision = self.hail_revision
+        version = self.hail_pip_version
+
+        filter_file(
+            r'\$\(error "git rev-parse HEAD" failed to produce output\)',
+            f"REVISION := {revision}",
+            "hail/version.mk",
+        )
+        filter_file(
+            r'\$\(error "git rev-parse --short=12 HEAD" failed to produce output\)',
+            f"SHORT_REVISION := {revision[:12]}",
+            "hail/version.mk",
+        )
+        filter_file(
+            r'\$\(error "git rev-parse --abbrev-ref HEAD" failed to produce output\)',
+            f"BRANCH := tags/{version}",
+            "hail/version.mk",
+        )
+
+        # Also need to make sure that build-info.properties gets the right revision
+        # which ends up improperly calculated in scala and will crash at runtime
+        filter_file(
+            r"val revision = VcsVersion\.vcsState\(\)\.currentRevision",
+            "val vcs_revision = VcsVersion.vcsState().currentRevision\n"
+            f'  val revision = if(vcs_revision ==  "no-vcs") "{revision}" else vcs_revision\n',
+            "hail/build.sc",
+        )
+
+    @property
+    def hail_revision(self):
+        version = self.version
+        version_info = self.versions[version]
+        # REVISION must look like a hash or Hail crashes at startup
+        # Technically, it needs to be at least 12 characters
+        revision = version_info.get("commit", version.joined.string.ljust(40, "0"))
+        return revision
+
+    @property
+    def hail_pip_version(self):
+        # This is the same behavior is as is defined in hail/version.mk
+        return f"{self.spec.version.up_to(3)}"
+
+    @property
+    def build_wheel_file_path(self):
+        wheel_file = f"hail-{self.hail_pip_version}-py3-none-any.whl"
+        wheel_dir = join_path("build", "deploy", "dist")
+        return join_path(wheel_dir, wheel_file)
+
+    def flag_handler(self, name, flags):
+        if name == "cxxflags":
+            # HAIL build doesn't find lz4: https://discuss.hail.is/t/ld-pruning-repeated-errors/1838/14
+            flags.append(f"-I{self.spec['lz4'].prefix.include}")
+        return (flags, None, None)
+
+    @property
+    def build_targets(self):
+        spec = self.spec
+
+        # Hail likes variables passed in to Make
+        variables = [
+            f"HAIL_PYTHON3={spec['python'].home.bin.python3}",
+            f"PIP={spec['py-pip'].home.bin.pip}",
+            f"SCALA_VERSION={spec['scala'].version}",
+            f"SPARK_VERSION={spec['spark'].version}",
+        ]
+        if spec.satisfies("+native"):
+            variables += ["HAIL_COMPILE_NATIVES=1"]
+
+        # We're not using the documented target to
+        # because it depends on pipto install and resolve
+        # dependencies directly and does everythin in one step.
```

Comment:
Some typos in here :slightly_smiling_face:

Reply:
:facepalm: